various-receipes/3rd-party-services-isolation/test/isolation.test.js

```diff
@@ -0,0 +1,118 @@
+const request = require("supertest");
+const sinon = require("sinon");
+const nock = require("nock");
+const {
+  initializeWebServer,
+  stopWebServer
+} = require("../../../example-application/api-under-test");
+const OrderRepository = require("../../../example-application/data-access/order-repository");
+
+let expressApp;
+let sinonSandbox;
+
+beforeAll(async (done) => {
+  // ️️️✅ Best Practice: Place the backend under test within the same process
+  expressApp = await initializeWebServer();
+  // ️️️✅ Best Practice: Ensure that this component is isolated by preventing unknown calls except for the Api-Under-Test
+  nock.disableNetConnect();
```

Comment:
Is disable needed? Wouldn't the next line, enableNetConnect, block all non-allowed requests?

Reply:
nope, see [docs](https://github.com/nock/nock#enabling-requests)